specs/discussions/2025-05-segments-global-history.md

```diff
@@ -0,0 +1,320 @@
+    pub final_child_segments: &'a [(ShardId, SegmentIndex)],
+   }
+   ```
+
+6. The beacon chain tracks the specific block heights from the intermediate shard at which
+   `child_segment_info` and `final_child_segments` were submitted, and will follow the same approach
+   used in intermediate shards to identify when a block can be considered final. Thus:
+
+   - Leaf shard segments will be included to the global history in the beacon chain if its parent
+     shard (and underlying intermediate shard) has submitted a `segment_info` and a subsequent
+     `final_child_segments` through the two corresponding `IntermediateBlockShardInfo` when the
+     finality has been reached, and once the block of the intermediate shard (parent shard) at which
+     the `final_child_segment` was included has also reached the `FINALITY_PROBABILITY_THRESHOLD` in
+     the beacon chain.
+   - Intermediate shard segments will be included to the global history in the beacon chain when the
+     the block of the intermediate shard where the `own_segment_info` for the shard has reached the
+     `FINALITY_PROBABILITY_THRESHOLD` in the beacon chain.
+   - Finally, beacon chain segments can be included into the global history and into a super segment
+     as soon as they are created (as any re-org in the beacon chain will impact the ordering of the
+     global history in any case).
+
+7. Once a new beacon chain block confirms the a set of segments from shards in the lower levels are
+   final, by their corresponding intermediate shard block reaching the
+   `FINALITY_PROBABILITY_THRESHOLD`, these segments and any beacon chain segments included in the
+   block are added to the global history, and a super segment is created from the segments in the
+   block. The super segment is created by computing the Merkle root of the tree of segments
+   aggregated in the super segment, and is included in the `super_segment_root` field of the
+   `SuperSegmentHeader`.
+   - To compute the super segment root, segments are ordered starting from the `own_segments_root`
+     of the beacon chain, and then in increasing order according to their shard ID and the
+     underlying block height in which the segments (or their confirmation) where included (e.g.
+     `own_segment_info_1`, `own_segment_info_2`, `intermediate_shard_segment_shard1`,
+     child_shard_segment_shard1`, `intermediate_shard_segment_shard2`, etc.).
+   - The super segment index is computed is determined by the number of segments aggregated in the
+     super segment, and the index of the super segment in the global history of the system. If the
+     previous super segment has `super_segment_index` 4 as it aggregates 4 super segments, if the
+     next super segments aggregates 5 segments, the `super_segment_index` of the next super segment
+     will be 4 + 5 = 9. In this way, the global history index for every segment of the shard is
+     assigned.
```

Comment:
This doesn't make a lot of sense to me. I expected that super segment would be increasing by `1` each time just like segment index or block number. For me this is a very non-intuitive way of doing it and it is not clear why it is needed or what it has to do with "global history index for every segment".